src/OrchardCore.Modules/OrchardCore.Workflows/Activities/SaveFormAttachmentsTask.cs

```diff
@@ -0,0 +1,99 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.FileStorage;
+using OrchardCore.FileStorage.FileSystem;
+using OrchardCore.Media;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Models;
+using OrchardCore.Workflows.Services;
+
+namespace OrchardCore.Workflows.Activities
+{
+    public class SaveFormAttachmentsTask : TaskActivity
+    {
+        readonly IOptions<ShellOptions> _shellOptions;
+        protected readonly IStringLocalizer S;
+        readonly ShellSettings _shellSettings;
+        private readonly IFileStore _fileStore;
+        private readonly IHttpContextAccessor _http;
+        private readonly IWorkflowExpressionEvaluator _expressionEvaluator;
+
+        public SaveFormAttachmentsTask(
+            IWorkflowExpressionEvaluator expressionEvaluator,
+            IStringLocalizer<SaveFormAttachmentsTask> localizer,
+            IOptions<ShellOptions> shellOptions,
+            ShellSettings shellSettings,
+            IMediaFileStore mediaFileStore,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            _shellOptions = shellOptions;
+            _shellSettings = shellSettings;
+            _fileStore = mediaFileStore;
+            _http = httpContextAccessor;
+            S = localizer;
+            _expressionEvaluator = expressionEvaluator;
+        }
+
+        public override string Name => nameof(SaveFormAttachmentsTask);
+
+        public override LocalizedString DisplayText => S["Save Form Attachments Task"];
+
+        public override LocalizedString Category => S["UI"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Done"]);
+        }
+
+        public override bool CanExecute(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return _http.HttpContext?.Request?.Form != null;
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            // use either configured mediaFileStore or create new FileSystemStore --> TODO: mediaFileStore does not resolve permissions, everything is in public folders and can be accessible.
+            // to put files in different folder, we would need either to register some private MediaFileStore, or allow to specify folder outside base path (?)
+            var fileStore = UseMediaFileStore ? _fileStore : CreateDefaultFileStore();
+            var filePaths = new List<string>();
+            foreach (var file in _http.HttpContext.Request.Form.Files)
+            {
+                var filePath = PathExtensions.Combine(Folder, $"{workflowContext.WorkflowId}-{file.FileName}");
```

Comment:
the funny thing is that I had same episode in my dev life, and wrote sth similiar to your code :P

maybe it would be better to use `_mediaFileStore.Combine` instead of `PathExtensions.Combine`

![image](https://user-images.githubusercontent.com/6403130/185373733-4bfe538b-66e0-4ccf-87f2-29470e24384a.png)

Reply:
@lampersky I guess there are not many other ways to implement this on OrchardCore framework, so it should be similliar :) 

If you have some code working, feel free to contribute and update this PR.

Anyway I have changed it to fileStore.Combine